modin/backends/pandas/query_compiler.py

```diff
@@ -2525,6 +2525,84 @@ def __convert_by(by):
 
         return unstacked
 
+    def pivot_table(
+        self,
+        index,
+        values,
+        columns,
+        aggfunc,
+        fill_value,
+        margins,
+        dropna,
+        margins_name,
+        observed,
+    ):
+        ErrorMessage.missmatch_with_pandas(
+            operation="pivot_table",
+            message="Order of columns could be different from pandas",
+        )
+
+        from pandas.core.reshape.pivot import _convert_by
+
+        def __convert_by(by):
+            if isinstance(by, pandas.Index):
+                return list(by)
+            return _convert_by(by)
+
+        index, columns, values = map(__convert_by, [index, columns, values])
+
+        unique_keys = np.unique(index + columns)
+        unique_values = np.unique(values)
+
+        if len(values):
+            to_group = self.getitem_column_array(unique_values)
+        else:
+            to_group = self.drop(columns=unique_keys)
+
+        keys_columns = self.getitem_column_array(unique_keys)
+
+        def applyier(df, other):
+            concated = pandas.concat([df, other], axis=1, copy=False)
+            result = concated.pivot_table(
+                index=index,
+                values=values if len(values) > 0 else None,
+                columns=columns,
+                aggfunc=aggfunc,
+                fill_value=fill_value,
+                margins=margins,
+                dropna=dropna,
+                margins_name=margins_name,
+                observed=observed,
+            )
+
+            # in that case Pandas transposes the result of `pivot_table`,
+            # transposing it back to be consistent with column axis values along
+            # different partitions
+            if len(index) == 0 and len(columns) > 0:
+                result = result.T
+
+            return result
+
+        result = self.__constructor__(
+            to_group._modin_frame.broadcast_apply_full_axis(
+                axis=0, func=applyier, other=keys_columns._modin_frame
+            )
+        )
+
+        # transposing the result again, to be consistent with Pandas result
+        if len(index) == 0 and len(columns) > 0:
+            result = result.transpose()
```

Comment:
Isn't the one transposing enough? As I see, we transpose result in this line and in `applyier` as well.

Reply:
We transpose result in `applier` to produce consistent partition. 

All partitions in a row must have the same indices and all partitions in a column must have the same columns. When we're applying our function to a columns partitions we're guarantee that partitions will be consistent. However with that conditions:
```python
if len(index) == 0 and len(columns) > 0
```
pandas transposes result of the `pivot_table` and frame loses it consistency:
```
before pandas transposing (consistent frame):
   a b |   c d
1  1 2 |1  7 8
2  3 4 |2  9 0
3  5 6 |3  1 2

after pandas transposing (inconsistent frame, because of different indices at the same row):
   1 2 3 |   1 2 3
a  1 3 5 |c  7 9 1
b  2 4 6 |d  8 0 2
```
so in `applyier` we transposing the result back to consistent state, and after that we're transposing the whole frame correctly in the query compiler